packages/vite/src/node/plugins/html.ts

```diff
@@ -230,7 +231,7 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
                   .map((child: any) => child.content || '')
                   .join('')
                 // <script type="module">...</script>
-                const filePath = id.replace(config.root, '')
+                const filePath = id.replace(normalizePath(config.root), '')
```

Comment:
On Windows, `config.root` has backslashes. But the following code assumes forward slashes only. So we call the `normalizePath()` utility.